src/formatDate.js

```diff
@@ -48,9 +48,46 @@
  *
  * @returns {string}
  */
-
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const newDate = [];
+  const oldSeparator = fromFormat[3];
+  const newSeparator = toFormat[3];
+  const divideDate = date.split(oldSeparator);
+  const dateMap = {};
+
+  for (let i = 0; i < divideDate.length; i++) {
+    dateMap[fromFormat[i]] = divideDate[i];
+  }
+
+  if (!dateMap['YY']) {
+    dateMap['YY'] = dateMap['YYYY'].slice(-2);
+  }
+
+  if (!dateMap['YYYY']) {
+    dateMap['YYYY'] = +dateMap['YY'] >= 30 ? `19${dateMap['YY']}` : `20${dateMap['YY']}`;
+  }
+
+  for (let i = 0; i <= 2; i++) {
+    switch (toFormat[i]) {
+      case 'DD':
+        newDate[i] = dateMap['DD'];
+        break;
+
+      case 'MM':
+        newDate[i] = dateMap['MM'];
+        break;
+
+      case 'YY':
+        newDate[i] = dateMap['YY'];
+        break;
+
+      default:
+        newDate[i] = dateMap['YYYY'];
+        break;
+    }
+  }
```

Comment:
You don't need to switch cases here. In all cases you do the same operation, but with different values. Let's simplify this and don't use any if or switch case clauses.

Hint: use `for in` loop to go though your `dateMap` object